smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/handlers/HttpQueryItemMiddleware.kt

```diff
@@ -46,22 +46,33 @@ class HttpQueryItemMiddleware(
         fun renderQueryMiddleware(ctx: ProtocolGenerator.GenerationContext, op: OperationShape, httpBindingResolver: HttpBindingResolver, defaultTimestampFormat: TimestampFormatTrait.Format) {
             val httpTrait = httpBindingResolver.httpTrait(op)
             val requestBindings = httpBindingResolver.requestBindings(op)
-            val queryBindings = requestBindings.filter { it.location == HttpBinding.Location.QUERY || it.location == HttpBinding.Location.QUERY_PARAMS }
+            val queryBindings =
+                requestBindings.filter { it.location == HttpBinding.Location.QUERY || it.location == HttpBinding.Location.QUERY_PARAMS }
             val queryLiterals = httpTrait.uri.queryLiterals
-
-            val inputSymbol = MiddlewareShapeUtils.inputSymbol(ctx.symbolProvider, ctx.model, op)
-            val outputSymbol = MiddlewareShapeUtils.outputSymbol(ctx.symbolProvider, ctx.model, op)
-            val outputErrorSymbol = MiddlewareShapeUtils.outputErrorSymbol(op)
-            val rootNamespace = MiddlewareShapeUtils.rootNamespace(ctx.settings)
-
-            val headerMiddlewareSymbol = Symbol.builder()
-                .definitionFile("./$rootNamespace/models/${inputSymbol.name}+QueryItemMiddleware.swift")
-                .name(inputSymbol.name)
-                .build()
-            ctx.delegator.useShapeWriter(headerMiddlewareSymbol) { writer ->
-                writer.addImport(SwiftDependency.CLIENT_RUNTIME.target)
-                val queryItemMiddleware = HttpQueryItemMiddleware(ctx, inputSymbol, outputSymbol, outputErrorSymbol, queryLiterals, queryBindings, defaultTimestampFormat, writer)
-                MiddlewareGenerator(writer, queryItemMiddleware).generate()
+            if (queryBindings.isNotEmpty() || queryLiterals.isNotEmpty()) {
```

Comment:
Similar to headers -- we should have this logic around whether we should/shouldn't render the middleware& rendering of intercept in an operation  in a single location. -- i think you wrote this, but it does not seem to be used.

Reply:
corrected